XamarinCommunityToolkit/Converters/ByteArrayToImageSourceConverter.shared.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Toolkit.Xamarin.Forms.Extensions;
+using System.Globalization;
+using Xamarin.Forms;
+using System.IO;
+using System;
+
+namespace Microsoft.Toolkit.Xamarin.Forms.Converters
+{
+    /// <summary>
+    /// Converts the incoming value from byte array and returns the object of a type ImageSource.
+    /// </summary>
+    public class ByteArrayToImageSourceConverter : ValueConverterExtension, IValueConverter
+    {
+        /// <summary>
+        /// Converts the incoming value from byte array and returns the object of a type ImageSource.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        /// <param name="targetType">The type of the binding target property.</param>
+        /// <param name="parameter">Additional parameter for the converter to handle. Not implemented.</param>
+        /// <param name="culture">The culture to use in the converter.</param>
+        /// <returns>An object of type ImageSource.</returns>
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value is null)
+                return default(ImageSource);
+
+            if (value is byte[] imageBytes)
+                return ImageSource.FromStream(() => new MemoryStream(imageBytes));
+
+            throw new ArgumentException("Expected value to be of type byte[].", nameof(value));
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value is null)
+                return null;
+
+            if (value is StreamImageSource streamImageSource)
+            {
+                var streamFromImageSource  = streamImageSource.Stream(System.Threading.CancellationToken.None).Result;
```

Comment:
```suggestion
                var streamFromImageSource = streamImageSource.Stream(System.Threading.CancellationToken.None).Result;
```

Reply:
This happens when you are in a hurry and want to finish things asap :D... thanks for the review! @AndreiMisiukevich